### PR TITLE
fixes following bugs in hr_hourly_cost,hr_timesheet,sale_timesheet

### PR DESCRIPTION
**Pull request by kais-odoo**
In this commit fixes below bugs which were produced in version 16,

- remove the 'per hour' label next to the hourly cost in employee form view
- minimize space on 'allocated hours' field input 

task-3000757

### CODE REVIEW

File: addons/hr_timesheet/views/project_sharing_views.xml [hunk 1]
```diff
@@ -22,12 +22,12 @@
                 <page string="Timesheets" id="timesheets_tab" attrs="{'invisible': [('allow_timesheets', '=', False)]}">
                     <group>
                         <group>
-                            <div class="o_td_label">
+                            <div colspan="2">
                                 <label for="planned_hours" string="Allocated Hours" class="me-2" attrs="{'invisible': [('encode_uom_in_days', '=', True)]}"/>
                                 <label for="planned_hours" string="Allocated Days" class="me-2" attrs="{'invisible': [('encode_uom_in_days', '=', False)]}"/>
-                                <field name="planned_hours" class="oe_inline" widget="timesheet_uom_no_toggle"/>
+                                <field name="planned_hours" class="timesheet_uom_no_toggle oe_inline" widget="timesheet_uom_no_toggle"/>
```
Comment (auon-odoo) [inline addons/hr_timesheet/views/project_sharing_views.xml:28]: https://github.com/odoo/odoo/blob/2a015600b2cf4631cbf16ca0802191c0f60c93d3/addons/web/static/src/views/form/form_controller.scss#L704
can't we use that?

**Review by auon-odoo — CHANGES_REQUESTED** [review]
Nice :fire: 
Only one question.

File: addons/web/static/src/views/fields/float_time/float_time_field.xml [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
     <t t-name="web.FloatTimeField" owl="1">
         <span t-if="props.readonly" t-esc="formattedValue" />
-        <input t-else="" t-att-id="props.id" t-ref="numpadDecimal" t-att-placeholder="props.placeholder" inputmode="numeric" class="o_input" />
+        <input t-else="" type="text" t-att-id="props.id" t-ref="numpadDecimal" t-att-placeholder="props.placeholder" inputmode="numeric" class="o_input" />
```
Comment (caburj) [inline addons/web/static/src/views/fields/float_time/float_time_field.xml:6]: Is this change really needed?

**Review by auon-odoo — APPROVED** [review]
Lgtm. Thank you! :fire:

File: addons/web/static/src/views/fields/float_time/float_time_field.xml (hunk 2 above)
Comment (xavierbol) [reply addons/web/static/src/views/fields/float_time/float_time_field.xml]: Hello @caburj , in fact it was to avoid a border when the user selects the input to edit it. 🙂

Comment (caburj) [reply addons/web/static/src/views/fields/float_time/float_time_field.xml]: Shoot. I should carefully judge even this very subtle change.

**Review by xavierbol — COMMENTED** [review]
Thanks for the fix, sorry for the delay, I was busy with the grid view. 😬 

Thanks for your work, I left one comment and after it will be good for me. 🙂

File: addons/web/static/src/views/fields/float_time/float_time_field.xml (hunk 2 above)
Comment (xavierbol) [reply addons/web/static/src/views/fields/float_time/float_time_field.xml]: I did the change in #115736 🙂

File: addons/hr_timesheet/views/project_views.xml [hunk 3]
```diff
@@ -130,10 +130,10 @@
...
                         <group>
                             <group>
-                                <label for="planned_hours" string="Allocated Hours" attrs="{'invisible': [('encode_uom_in_days', '=', True)]}"/>
-                                <label for="planned_hours" string="Allocated Days" attrs="{'invisible': [('encode_uom_in_days', '=', False)]}"/>
-                                <div class="o_row">
-                                    <field name="planned_hours" class="oe_inline ms-2" widget="timesheet_uom_no_toggle"/>
+                                <div colspan="2">
+                                    <label for="planned_hours" string="Allocated Hours" attrs="{'invisible': [('encode_uom_in_days', '=', True)]}"/>
+                                    <label for="planned_hours" string="Allocated Days" attrs="{'invisible': [('encode_uom_in_days', '=', False)]}"/>
+                                    <field name="planned_hours" class="o_field_float_time oe_inline ms-2" widget="timesheet_uom_no_toggle"/>
```
[the remark this answers is not in the document]
Comment (kais-odoo) [reply addons/hr_timesheet/views/project_views.xml]: Hello @xavierbol done the changes. :slightly_smiling_face:

File: addons/hr_timesheet/views/project_sharing_views.xml [hunk 4]
```diff
@@ -22,12 +22,12 @@
...
                     <group>
                         <group>
-                            <div class="o_td_label">
-                                <label for="planned_hours" string="Allocated Hours" class="me-2" attrs="{'invisible': [('encode_uom_in_days', '=', True)]}"/>
-                                <label for="planned_hours" string="Allocated Days" class="me-2" attrs="{'invisible': [('encode_uom_in_days', '=', False)]}"/>
-                                <field name="planned_hours" class="oe_inline" widget="timesheet_uom_no_toggle"/>
+                            <label for="planned_hours" string="Allocated Hours" attrs="{'invisible': [('encode_uom_in_days', '=', True)]}"/>
+                            <label for="planned_hours" string="Allocated Days" attrs="{'invisible': [('encode_uom_in_days', '=', False)]}"/>
+                            <div class="o_row">
+                                <field name="planned_hours" class="o_field_float_time oe_inline ms-2" widget="timesheet_uom_no_toggle"/>
```
Comment (xavierbol) [inline addons/hr_timesheet/views/project_sharing_views.xml:28]: This diff can be done in master but not in master, I suggest to put your last changes instead and in master, you will do that. 🙂 

Because here you changes the xml structure and so you could break the customer customisation (xpath).

Comment (kais-odoo) [reply addons/hr_timesheet/views/project_sharing_views.xml]: yes, that's correct. my bad :see_no_evil:

**Review by xavierbol — APPROVED** [review]
LGTM, thanks for your work. 🙂 

robodoo r+

**Comment by xavierbol** [comment]
robodoo rebase-ff
